staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -93,6 +93,9 @@ func SetOldTransportDefaults(t *http.Transport) *http.Transport {
 	if t.Dial == nil {
 		t.Dial = defaultTransport.Dial
 	}
+	if t.DialContext == nil {
+		t.DialContext = defaultTransport.DialContext
```

Comment:
Is it correct to set both Dial and DialContext? If Dial was explicitly set, would setting DialContext from the default transport override it? If so, that isn't what we want

Reply:
Yeah, I think it should be like this, defaultTransport.Dial is nil now.
```
+       if t.Dial == nil && t.DialContext == nil {
+               t.DialContext = defaultTransport.DialContext
        }

```